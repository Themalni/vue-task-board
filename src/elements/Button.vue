<template>
  <div>
    <button class="button" :class="[ color ]" @click="$emit('addAction')">
      <slot></slot>
    </button>
  </div>
</template>

<script>
export default {
  name: "Button",
  props: {
    color: {
      type: String
    }
  },
  methods: {
    addAction() {
      this.$emit("addAction");
    }
  }
};
</script>

<style lang="scss">
.button {
  padding: 0.7em 1.5em;
  border: none;
  border-radius: 2px;
  box-shadow: 0 1px 5px -3px $graphite;
  cursor: pointer;
  transition: 0.2s linear;

  &:hover {
    box-shadow: 0px 3px 8px -5px $graphite;
    transition: 0.2s linear;
  }
}
.button-default {
  background-color: $light-grey;
  color: $graphite;

  &:hover {
    background-color: darken($light-grey, 5%);
    transition: 0.2s linear;
  }
}
.button-primary {
  background-color: $blue;
  color: $white;

  &:hover {
    background-color: darken($blue, 5%);
    transition: 0.2s linear;
  }
}
.button-success {
  background-color: $olive;
  color: $white;

  &:hover {
    background-color: darken($olive, 5%);
    transition: 0.2s linear;
  }
}
.button-danger {
  background-color: $crimson;
  color: $white;

  &:hover {
    background-color: darken($crimson, 5%);
    transition: 0.2s linear;
  }
}
</style>