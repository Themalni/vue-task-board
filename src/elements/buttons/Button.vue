<template>
  <div>
    <button class="button" :class="[ color ]" @click="$emit('addAction')" :disabled="disabled">
      <slot></slot>
    </button>
  </div>
</template>

<script>
export default {
  name: 'Button',
  props: {
    color: {
      type: String
    },
    disabled: {
      type: Boolean
    }
  },
  methods: {
    addAction () {
      this.$emit('addAction')
    }
  }
}
</script>

<style lang="scss">
.button {
  padding: 0.7em 1.5em;
  border: none;
  border-radius: 2px;
  box-shadow: 0 1px 5px -3px $graphite;
  cursor: pointer;
  transition: 0.2s linear;

  &:hover {
    box-shadow: 0px 3px 8px -5px $graphite;
    transition: 0.2s linear;
  }

  &:disabled {
    pointer-events: none;
  }
}
.button-default {
  background-color: $light-grey;
  color: $graphite;

  &:hover {
    background-color: darken($light-grey, 5%);
    transition: 0.2s linear;
  }

  &:disabled {
    background-color: lighten($light-grey, 25%);
  }
}
.button-primary {
  background-color: $blue;
  color: $white;

  &:hover {
    background-color: darken($blue, 5%);
    transition: 0.2s linear;
  }

  &:disabled {
    background-color: lighten($blue, 25%);
  }
}
.button-success {
  background-color: $olive;
  color: $white;

  &:hover {
    background-color: darken($olive, 5%);
    transition: 0.2s linear;
  }

  &:disabled {
    background-color: lighten($olive, 25%);
  }
}
.button-danger {
  background-color: $crimson;
  color: $white;

  &:hover {
    background-color: darken($crimson, 5%);
    transition: 0.2s linear;
  }
}
</style>
