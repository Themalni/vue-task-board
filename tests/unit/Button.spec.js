import { mount } from '@vue/test-utils'
import Button from '@/elements/buttons/Button'

let wrapper
let button

beforeEach(() => {
  wrapper = mount(Button)
  button = wrapper.find('button')
})

afterEach(() => {
  wrapper.destroy()
})

describe('Button', () => {
  it('does contain `button` element', () => {
    expect(button.is('button')).toBe(true)
  })
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('does not set button as disabled by default', () => {
    expect(button.attributes('disabled')).not.toBe('disabled')
  })
  it('does not set button as disabled by default', () => {
    const propsData = {
      disabled: true
    }
    wrapper.setProps(propsData)
    expect(button.attributes('disabled')).toBe('disabled')
  })
  it('does add class `button-primary` when value is provided to prop `color`', () => {
    const propsData = {
      color: 'button-primary'
    }
    wrapper.setProps(propsData)
    expect(button.classes('button-primary')).toBeTruthy()
  })
  it('emits event `addAction`', () => {
    wrapper.vm.addAction()
    expect(wrapper.emitted().addAction).toBeTruthy()
  })
})
