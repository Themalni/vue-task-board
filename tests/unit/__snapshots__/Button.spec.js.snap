// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renters correctly 1`] = `<div><button class="button"></button></div>`;
